test/extended/operators/server_side_apply_examples.go

```diff
@@ -0,0 +1,93 @@
+	exutil "github.com/openshift/origin/test/extended/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+var _ = g.Describe("[sig-apimachinery]", func() {
+
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLIWithPodSecurityLevel("server-side-apply-examples", admissionapi.LevelPrivileged)
+	fieldManager := metav1.ApplyOptions{
+		FieldManager: "e2e=test",
+	}
+
+	g.Describe("server-side-apply should function properly", func() {
+		g.It("should clear fields when they are no longer being applied", func() {
+			ctx := context.Background()
+
+			// TODO use another API.  I might have a version in my cache somewhere using a pod.
+			_, err := oc.AdminConfigClient().ConfigV1().ClusterOperators().Create(ctx, &configv1.ClusterOperator{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-instance",
+				},
+			}, metav1.CreateOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			defer oc.AdminConfigClient().ConfigV1().ClusterOperators().Delete(ctx, "test-instance", metav1.DeleteOptions{})
+
+			addFirstCondition := applyconfigv1.ClusterOperator("test-instance").
+				WithStatus(applyconfigv1.ClusterOperatorStatus().
+					WithConditions(applyconfigv1.ClusterOperatorStatusCondition().
+						WithType("FirstType").
+						WithStatus(configv1.ConditionTrue).
+						WithReason("Dummy").
+						WithMessage("No Value").
+						WithLastTransitionTime(metav1.Now()),
+					),
+				)
+			_, err = oc.AdminConfigClient().ConfigV1().ClusterOperators().ApplyStatus(ctx, addFirstCondition, fieldManager)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			currInstance, err := oc.AdminConfigClient().ConfigV1().ClusterOperators().Get(ctx, "test-instance", metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			if !containsCondition(currInstance.Status.Conditions, "FirstType") {
+				framework.Logf("got conditions: %v", spew.Sdump(currInstance.Status.Conditions))
+				g.Fail("missing FirstType condition")
+			}
+
+			addJustSecondCondition := applyconfigv1.ClusterOperator("test-instance").
+				WithStatus(applyconfigv1.ClusterOperatorStatus().
+					WithConditions(applyconfigv1.ClusterOperatorStatusCondition().
+						WithType("SecondType").
+						WithStatus(configv1.ConditionTrue).
+						WithReason("Dummy").
+						WithMessage("No Value").
+						WithLastTransitionTime(metav1.Now()),
+					),
+				)
+			_, err = oc.AdminConfigClient().ConfigV1().ClusterOperators().ApplyStatus(ctx, addJustSecondCondition, fieldManager)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			currInstance, err = oc.AdminConfigClient().ConfigV1().ClusterOperators().Get(ctx, "test-instance", metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			if !containsCondition(currInstance.Status.Conditions, "SecondType") {
+				g.Fail("missing SecondType condition")
+			}
+			if containsCondition(currInstance.Status.Conditions, "FirstType") {
+				g.Fail("has FirstType condition unexectedly")
```

Comment:
notice this line doesn't execute because the condition is not present.

Reply:
ErrUnexpectedTypo
```suggestion
				g.Fail("has FirstType condition unexpectedly")
```